### DISCUSSION TITLE
How to upload file /image with WebdriverIO by interacting with system window when there is no input tag available in DOM

### DISCUSSION THREAD

**Discussion by MOLMHQ** [webdriverio/webdriverio#8470] [Q&A]
Steps to reproduce: 
Note: We have input filed in DOM, that is visible only when we click on upload button, but at the same time system window will get open.

1. We have to click on upload button in browser.
2.  Then system window will be displayed, 
3.  Need to pass the file source path in system window  and click on open button in it.

![uploadfile](https://user-images.githubusercontent.com/108267239/175909792-3b20da7c-2032-4641-b2be-c673f8de6e06.png)

**Comment by christian-bromann** [comment]
Please see the [`uploadFile`](https://webdriver.io/docs/api/browser/uploadFile/) command.

**Reply by MOLMHQ** [reply]
Thanks @christian-bromann 

We tried that and it didn't work for us, as our scenario is different as mentioned

we want to input our source file path in the system input dialogue box.

**Reply by christian-bromann** [reply]
You can't automate system dialogs. Why would you? These things are tested by the OS, you should focus the automation on your application.

**Reply by MOLMHQ** [reply]
Our application opens system dialogs to take the upload file path from local.

My concern is, the hidden input tag is displayed only after clicking the upload button.

We are using this library for upload functionality in our application https://docs.expo.dev/versions/latest/sdk/document-picker/

**Reply by christian-bromann** [reply]
Then make sure that this upload library is well tested. It is unfortunately not possible to automate system dialogs.

**Comment by SivasubramanianV** [comment]
@MOLMHQ Were you able to solve this situation, I am also not able to upload files but I am using a mac.
